spec.html

```diff
@@ -17935,7 +17935,7 @@ <h1>Runtime Semantics: ForIn/OfHeadEvaluation ( _uninitializedBoundNames_, _expr
             1. Let _obj_ be ! ToObject(_exprValue_).
             1. Return ? EnumerateObjectProperties(_obj_).
           1. Else,
-            1. Assert: _iterationKind_ is ~iterate~.
+            1. Assert: _iterationKind_ is ~iterate~ or ~async-iterate~.
```

Comment:
given that the immediately following line checks if it's `~async-iterate~`, this seems like a clear bug fix to me.